HIP/hip-1340.md

```diff
@@ -0,0 +1,388 @@
+---
+hip: 1340
+title: EOA Code Delegation
+author: Lukasz Gasior (@lukasz.gasior), Luke Lee (@luke.lee)
+working-group: TBD
+type: Standards Track
+category: Service
+needs-hiero-approval: Yes 
+needs-hedera-approval: Yes
+status:  Draft
+created: 2025-11-13
+updated: 2025-11-13
+---
+
+## Abstract
+
+Ethereum's Pectra upgrade (activated May 7, 2025) introduced, among other things, EOA (Externally Owner Account) Code Delegation. It has originally been defined in EIP-7702. This HIP defines the adoption of this feature in Hiero, Hiero-specific modifications and extensions of it, and the related changes to HSCS's proxy/facade mechanism.
+
+## Motivation
+
+EOA Code Delegation is the core of Ethereum's Pectra upgrade and enables a wide range of use cases. Supporting it aligns with the goal of maintaining compatibility with Ethereum Mainnet.
+
+## Rationale
+
+The main proposal of this HIP is to implement EOA Code Delegation closely mimicking the original specification defined in EIP-7702.
+
+All proposed changes are a result of analysis of how Code Delegation can best fit into the existing protocol.
+
+## User Stories
+
+- As a network user I want to be able to set Code Delegation for an EOA using Ethereum-comptabile wallets and tools, so that EVM calls to my account can result in code execution.
+- As a network user I want to be able to set Code Delegation for an EOA using native `CryptoCreate` or `CryptoUpdate` transactions.
+- As a network user I want to be able to call an EOA and execute a correct delegate Smart Contract code if Code Delegation has been set for that account.
+- As a network user I want to be able to use the existing HTS/HSS/HAS facade calls with no interface changes.
+
+## Specification
+
+EOA Code Delgation has originally been defined in EIP-7702. We propose to adopt it with minimum changes.
+
+### High-Level Summary
+
+The primary feature of EOA Code Delegation is to allow users to configure their EOAs to point to an EVM code stored in another on-chain Smart Contract account.
+When such a configuration is set up, any EVM calls targetting that account will result
+in execution of the Smart Contract's EVM code in the context of original EOA (similarly to how DELEGATECALL executes the code of Contract B in the context of Contract A).
+
+![EOA Code Delegation High-Level](../assets/hip-1340/eoa_code_delegation_high_level.png)
```

Comment:
This diagram does not make clear that though using the delegate contract's bytecode it maintains its own separate EOA storage. It's mentioned elsewhere but not visually.